sources/nvidia-migmanager/src/main.rs

```diff
@@ -0,0 +1,688 @@
+                    (ampere || is_ampere, disabled || is_disabled)
+                });
+
+        if has_disabled_mig {
+            // Enable MIG for all the GPU
+            set_mig_mode(true)?;
+
+            // If any GPU is A100 create marker file for reboot to reconcile
+            // for the gpu reset and move from transitional state to enabled
+            if is_ampere_gpu_present {
+                info!("Rebooting to apply MIG Settings.");
+                std::fs::write("/run/nvidia-migmanager/reboot-required", "Enabling MIG").unwrap();
+            }
+        }
+
+        let a100_40gb_indices: Vec<usize> = gpu_info
+            .iter()
+            .enumerate()
+            .filter(|(_, gpu)| gpu.model == NvidiaGpu::A100_40GB)
+            .map(|(index, _)| index)
+            .collect();
+        process_mig_config::<NvidiaA100_40gbMigProfile>(
+            "a100.40gb",
+            a100_40gb_indices,
+            &mig_settings,
+        )?;
+
+        let a100_80gb_indices: Vec<usize> = gpu_info
+            .iter()
+            .enumerate()
+            .filter(|(_, gpu)| gpu.model == NvidiaGpu::A100_80GB)
+            .map(|(index, _)| index)
+            .collect();
+        process_mig_config::<NvidiaA100_80gbMigProfile>(
+            "a100.80gb",
+            a100_80gb_indices,
+            &mig_settings,
+        )?;
+
+        let h100_80gb_indices: Vec<usize> = gpu_info
+            .iter()
+            .enumerate()
+            .filter(|(_, gpu)| gpu.model == NvidiaGpu::H100_80GB)
+            .map(|(index, _)| index)
+            .collect();
+        process_mig_config::<NvidiaH100_80gbMigProfile>(
+            "h100.80gb",
+            h100_80gb_indices,
+            &mig_settings,
+        )?;
+
+        let h200_141gb_indices: Vec<usize> = gpu_info
+            .iter()
+            .enumerate()
+            .filter(|(_, gpu)| gpu.model == NvidiaGpu::H200_141GB)
+            .map(|(index, _)| index)
+            .collect();
+        process_mig_config::<NvidiaH200_141gbMigProfile>(
+            "h200.141gb",
+            h200_141gb_indices,
+            &mig_settings,
+        )?;
+
+        let other_indices: Vec<usize> = gpu_info
+            .iter()
+            .enumerate()
+            .filter(|(_, gpu)| gpu.model == NvidiaGpu::Other)
+            .map(|(index, _)| index)
+            .collect();
+        let known_gpus: Vec<NvidiaGpuModel> = vec![
+            NvidiaGpuModel::try_from("a100.40gb").unwrap(),
+            NvidiaGpuModel::try_from("a100.80gb").unwrap(),
+            NvidiaGpuModel::try_from("h100.80gb").unwrap(),
+            NvidiaGpuModel::try_from("h200.141gb").unwrap(),
+        ]
+        .into_iter()
+        .collect();
+
+        // The GPU in the current instance is not one of the known GPUs. We attempt using a profile that doesn't belong to one of the known GPUs.
+        for (gpu, mig_profile) in &mig_settings.profile {
+            if !known_gpus.contains(gpu) {
```

Comment:
Isn't this always true? You could make it another `ensure!` or omit the check, to help de-indent the code.